### PR TITLE
HTTPS sensitive URIs

### LINKED ISSUE #2960: Access Nominatim via HTTPS for location search
When trying to do a location search using the "Search worldwide" button, my browser is blocking the request because I loaded osm.org via HTTPS but iD is trying to contact Nominatim via HTTP. I am using Chromium but I think all browsers have become more strict about this in the last year or two.

I see for reverse geocoding Nominatim's URL was changed in 16a2449 but the search feature is still pointing at HTTP. I believe the simplest fix might be to just change [line 248 in feature_list.js](https://github.com/openstreetmap/iD/blob/master/js/id/ui/feature_list.js#L248) to point at https://

### PR DESCRIPTION
**Pull request by kepta**
change of all http:// URIs in js/id folder to //.

(closes #2960)

### CODE REVIEW

File: js/id/renderer/background_source.js [hunk 1]
```diff
@@ -117,7 +117,7 @@ iD.BackgroundSource.Bing = function(data, dispatch) {
     };
 
     bing.logo = 'bing_maps.png';
-    bing.terms_url = 'http://opengeodata.org/microsoft-imagery-details';
+    bing.terms_url = '//opengeodata.org/microsoft-imagery-details';
```
Comment (bhousel) [inline js/id/renderer/background_source.js]: Looks like this has moved to:
https://blog.openstreetmap.org/2010/11/30/microsoft-imagery-details/

File: js/id/ui/splash.js [hunk 2]
```diff
@@ -23,8 +23,8 @@ iD.ui.Splash = function(context) {
             .append('p')
             .html(t('splash.text', {
                 version: iD.version,
-                website: '<a href="http://ideditor.com/">ideditor.com</a>',
-                github: '<a href="https://github.com/openstreetmap/iD">github.com</a>'
+                website: '<a href="//ideditor.com/">ideditor.com</a>',
```
Comment (bhousel) [inline js/id/ui/splash.js]: This one isn't available over https
(we should really just move it to github pages)

**Comment by bhousel** [comment]
see also discussion on #2681 

For outbound links and api endpoints, we need to make sure the resource is available over both `http` and `https`.  I commented in the diff the places where that's not true.

Given that iD can sometimes be loaded locally, and `file://` urls can break in surprising ways, maybe we should instead add the `location.protocol === 'http' ? 'http' : 'https'` check to `iD.detect()`?

Thoughts @jfirebaugh ?

**Comment by kepta** [comment]
This `//` will load the hyperlink depending on the iD's current connection status 
if plain then // will go to http://
if encypted then // will go to https://

**Comment by jfirebaugh** [comment]
Please don't use protocol relative URLs -- as @bhousel said they fail for `file://`, and https is preferred even when loading the referring page on http. Therefore:

* If the site supports https, use https unconditionally.
* If not, look for an alternative URL.
* Failing that, bug the site owner for https support.
* Failing that, stick to http.

**Comment by kepta** [comment]
@bhousel I have updated the patch.
Please have a look.

**Comment by bhousel** [comment]
Thanks this looks good!

**Pull request merged**